vault/ui_custom_messages/manager.go

```diff
@@ -0,0 +1,236 @@
+		// If no error occured but the specified message doesn't exist...
+		return nil, nil
+	}
+
+	if err = m.putEntry(ctx, entry); err != nil {
+		return nil, err
+	}
+
+	return &message, nil
+}
+
+// DeleteMessage handles deleting the message with the provided id value in the
+// current namespace if it exists. The method updates the logical.Storage as
+// well.
+func (m *Manager) DeleteMessage(ctx context.Context, id string) error {
+	m.l.Lock()
+	defer m.l.Unlock()
+
+	entry, err := m.getEntry(ctx)
+	if err != nil {
+		return err
+	}
+
+	delete(entry.Messages, id)
+
+	return m.putEntry(ctx, entry)
+}
+
+// getEntry is a helper method that retrieves the current namespace from the
+// context.Context and uses it to call getEntryForNamespace.
+func (m *Manager) getEntry(ctx context.Context) (*Entry, error) {
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return m.getEntryForNamespace(ctx, ns)
+}
+
+// getEntryForNamespace takes care of retrieving the logical.StorageEntry that
+// corresponds to the provided namespace.Namespace. The logical.StorageEntry is
+// then used to build an Entry struct.
+func (m *Manager) getEntryForNamespace(ctx context.Context, ns *namespace.Namespace) (*Entry, error) {
+	if ns == nil {
+		return nil, errors.New("missing namespace")
+	}
+
+	storageEntry, err := m.view.Get(ctx, ns.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if storageEntry == nil {
+		return &Entry{
+			Messages: make(map[string]Message),
+		}, nil
+	}
+
+	var entry *Entry = new(Entry)
+	if err := storageEntry.DecodeJSON(entry); err != nil {
+		return nil, err
+	}
+
+	return entry, nil
+}
+
+// putEntry takes care of determining the current namespace from the
+// context.Context and then marshalling the provided Entry pointer to a slice
+// of byte to create the appropriate logical.StorageEntry and then storing it
+// in the logical.Storage.
+func (m *Manager) putEntry(ctx context.Context, entry *Entry) error {
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	value, err := json.Marshal(entry)
+	if err != nil {
+		return err
+	}
+
+	storageEntry := &logical.StorageEntry{
+		Key:   ns.ID,
+		Value: value,
+	}
+
+	return m.view.Put(ctx, storageEntry)
+}
+
+// getNamespacesToSearch builds a slice of pointers to namespace.Namespace
+// struct that will be walked by the (*Manager).FindMessage method above.
+// This function handles the complexity of gathering all of the applicable
+// namespaces depending on the namespace set in the context and whether the
+// IncludeAncestors criterion is set to true in the provided FindFilter struct.
+func getNamespacesToSearch(ctx context.Context, filters FindFilter) ([]*namespace.Namespace, error) {
+	var nsList []*namespace.Namespace
+
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add the current namespace based on the context.Context to nsList.
+	nsList = append(nsList, ns)
+
+	//if filters.IncludeAncestors {
```

Comment:
Is this code meant to be deleted, or implemented? I think based on the comment, implemented

Reply:
Ah! Makes sense. In the future, feel free to put that explicitly in a comment (as long as you remove the comment in the future implementation!) Saves the back and forth in review :)